qutip/core/qobj.py

```diff
@@ -512,9 +512,7 @@ def __matmul__(self, other):
                     copy=False)
 
     def __truediv__(self, other):
-        if not isinstance(other, numbers.Number):
-            return NotImplemented
-        return self.__mul__(1 / complex(other))
+        return self.__mul__(1 / other)
```

Comment:
Do we need a `try: ... except TypeError: return NotImplemented` here like we did in #1620 for `__mul__`?

Reply:
Thats a good point. With the current implementation it will return `NotImplemented` when doing `mul` so I assume it will default to not trying `(1/other).__mul__`. Not terrible but maybe not ideal. Should I implement it to default to `other.__truediv__`?

Reply:
Well, it turns out that the `NotImplemented` returned by `__mul__` should already work. I added some new tests to ensure that is the case.